src/coreclr/vm/class.cpp

```diff
@@ -2702,8 +2702,18 @@ void EEClass::AddChunk (MethodDescChunk* pNewChunk)
     STATIC_CONTRACT_FORBID_FAULT;
 
     _ASSERTE(pNewChunk->GetNextChunk() == NULL);
-    pNewChunk->SetNextChunk(GetChunks());
-    SetChunks(pNewChunk);
+
+    MethodDescChunk* head =  GetChunks();
+
+    if (head == NULL)
+        SetChunks(pNewChunk);
+    else
+    {
```

Comment:
It may be useful to add a comment here or at the callsite that reflection depends on the chunk order.